runtime/src/bank/partitioned_epoch_rewards/calculation.rs

```diff
@@ -95,6 +94,26 @@ impl Bank {
         thread_pool: &ThreadPool,
         metrics: &mut RewardsMetrics,
     ) -> CalculateRewardsAndDistributeVoteRewardsResult {
+        let mut epoch_rewards_calculation_cache =
```

Comment:
This approach means that we can no longer calculate epoch rewards for competing forks in parallel. If they have the same parent hash that's great, but if they have different parent hashes we could actually make replay slower. I think we only want to lock based on parent hash. Thoughts? cc @brooksprumo

Reply:
Hmm actually `generate_new_bank_forks` only creates new banks in serial right now. This would only be an issue if we decided to start creating new banks in parallel. So this lock is actually not going to have any contention at all right now.

Reply:
Yes, all banks are replayed sequentially for now. 
In future, if we ever change to replay bank in parallel, we could revisit and change the cache to dashmap when needed.

Reply:
Another question to ask is whether we want to process the forks in parallel ever at epoch boundary?

If we process forks in parallel, they would be competing for the computing resources and slow down the PER for both the forks, which will slow down the votes and consensus for the first fork, the most likely fork to root. So, it seems
that it is better to process forks sequentially at epoch boundary, i.e. lock and let other forks wait for the first fork to finish computation before they can start. Does this make sense?